pkg/controller/restore_session.go

```diff
@@ -298,7 +298,16 @@ func (c *StashController) ensureRestoreJob(restoreSession *api_v1beta1.RestoreSe
 		jobTemplate.Spec.Volumes = core_util.UpsertVolume(jobTemplate.Spec.Volumes, volumes...)
 
 		// add ordinal suffix to the job name so that multiple restore job can run concurrently
-		jobTemplate.Name = fmt.Sprintf("%s-%d", jobMeta.Name, ordinal)
+		for i, c := range jobTemplate.Spec.Containers {
+			if c.Name == util.StashContainer {
+				for j, e := range jobTemplate.Spec.Containers[i].Env {
+					if e.Name == util.KeyPodName {
```

Comment:
Why special case this? Why are we not using Kubernetes downward api?

https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-pod-fields-as-values-for-environment-variables

Reply:
Downward API is not suitable for this scenario.
But Using this API, created pod suffix is a random string. It's not expected.  If we use this, the resolved host value is of the form `host-<last_part_of_the_Pod_name>`.